packages/contracts-bedrock/test/DataAvailabilityChallenge.t.sol

```diff
@@ -0,0 +1,496 @@
+        dac.challenge(challengedBlockNumber, challengedHash);
+
+        // Move to a block after the challenge window
+        vm.roll(block.number + dac.resolveWindow() + 1);
+
+        // Unlock the bond
+        dac.unlockBond(challengedBlockNumber, challengedHash);
+
+        uint256 balanceAfterUnlock = dac.balances(address(this));
+
+        // Unlock the bond again doesn't change the balance
+        dac.unlockBond(challengedBlockNumber, challengedHash);
+        assertEq(dac.balances(address(this)), balanceAfterUnlock);
+    }
+
+    function testUnlockFailResolveWindowNotClosed() public {
+        bytes memory preImage = "some preimage";
+        bytes32 challengedHash = keccak256(preImage);
+        uint256 challengedBlockNumber = 1;
+
+        // Move to block after challenged block
+        vm.roll(challengedBlockNumber + 1);
+
+        // Challenge the hash
+        dac.deposit{ value: dac.bondSize() }();
+        dac.challenge(challengedBlockNumber, challengedHash);
+
+        vm.roll(block.number + dac.resolveWindow() - 1);
+
+        // Expiring the challenge before the resolve window closes fails
+        vm.expectRevert(abi.encodeWithSelector(DataAvailabilityChallenge.ChallengeNotExpired.selector));
+        dac.unlockBond(challengedBlockNumber, challengedHash);
+    }
+
+    function testSetBondSize() public {
+        uint256 requiredBond = dac.bondSize();
+        uint256 actualBond = requiredBond - 1;
+        dac.deposit{ value: actualBond }();
+
+        // Expect the challenge to fail because the bond is too low
+        vm.expectRevert(abi.encodeWithSelector(DataAvailabilityChallenge.BondTooLow.selector, actualBond, requiredBond));
+        dac.challenge(0, "some hash");
+
+        // Reduce the required bond
+        vm.prank(DAC_OWNER);
+        dac.setBondSize(actualBond);
+
+        // Expect the challenge to succeed
+        dac.challenge(0, "some hash");
+    }
+
+    function testSetResolverRefundPercentage(uint256 resolverRefundPercentage) public {
+        vm.assume(resolverRefundPercentage <= 100);
+        vm.prank(DAC_OWNER);
+        dac.setResolverRefundPercentage(resolverRefundPercentage);
+        assertEq(dac.resolverRefundPercentage(), resolverRefundPercentage);
+    }
+
+    function testSetResolverRefundPercentageFail() public {
+        vm.expectRevert(abi.encodeWithSelector(DataAvailabilityChallenge.InvalidResolverRefundPercentage.selector, 101));
+        vm.prank(DAC_OWNER);
+        dac.setResolverRefundPercentage(101);
+    }
+
+    function testSetBondSizeFailOnlyOwner(address notOwner, uint256 newBondSize) public {
+        vm.assume(notOwner != DAC_OWNER);
+
+        // Expect setting the bond size to fail because the sender is not the owner
+        vm.prank(notOwner);
+        vm.expectRevert("Ownable: caller is not the owner");
+        dac.setBondSize(newBondSize);
+    }
+}
+
+contract DataAvailabilityChallengeProxyTest is DataAvailabilityChallengeTest {
```

Comment:
This seems to be not used

Reply:
It runs every test defined above in the proxy context (where the DAC is not interacted with directly but via the proxy) - might not be necessary if you're confident in the proxy though :)

Reply:
removed the non-proxy test